wcfsetup/install/files/lib/data/search/keyword/SearchKeywordAction.class.php

```diff
@@ -56,4 +56,21 @@ public function getSearchResultList() {
 		
 		return $list;
 	}
+	
+	/**
+	 * Inserts a new keyword if it does not already exist, or updates it if it does.
+	 */
+	public function upsert() {
```

Comment:
```suggestion
	public function registerSearch() {
```

`upsert` is awfully generic.